imhamt/src/lib.rs

```diff
@@ -139,7 +140,7 @@ mod tests {
     }
 
     #[allow(clippy::trivially_copy_pass_by_ref)]
-    fn next_u32(x: &u32) -> Result<Option<u32>, ()> {
+    fn next_u32(x: &u32) -> Result<Option<u32>, Infallible> {
```

Comment:
clippy is right by the way, it's pointless to pass an integer by reference.

Reply:
Yeah, it's right, but this function is used _only_ for `Hamt::update` and it passes the current value by reference.

Reply:
I think we should have been less liberal with throwing those clippy overrides around, even if they only relate to internal code. Nothing major here, though.